packages/keyring-eth-ledger-bridge/src/ledger-mobile-bridge.test.ts

```diff
@@ -155,50 +155,57 @@ describe('LedgerMobileBridge', function () {
       const hdPath = "m/44'/60'/0'/0/0";
       const tx =
         'f86d8202b38477359400825208944592d8f8d7b001e72cb26a73e4fa1806a51ac79d880de0b6b3a7640000802ba0699ff162205967ccbabae13e07cdd4284258d46ec1051a70a51be51ec2bc69f3a04e6944d508244ea54a62ebf9a72683eeadacb73ad7c373ee542f1998147b220e';
-      const resolution = await ledgerService.resolveTransaction(tx, {}, {});
       await bridge.deviceSignTransaction({
         hdPath,
         tx,
       });
       expect(transportMiddlewareGetEthAppSpy).toHaveBeenCalledTimes(1);
-      expect(mockEthApp.signTransaction).toHaveBeenCalledTimes(1);
-      expect(mockEthApp.signTransaction).toHaveBeenCalledWith(
-        hdPath,
-        tx,
-        resolution,
-      );
+      expect(mockEthApp.clearSignTransaction).toHaveBeenCalledTimes(1);
+      expect(mockEthApp.clearSignTransaction).toHaveBeenCalledWith(hdPath, tx, {
+        externalPlugins: true,
+        erc20: true,
+        nft: true,
+      });
     });
 
-    it('throws an error when tx format is not correct', async function () {
+    it('returns undefined when tx format is not correct', async function () {
       const hdPath = "m/44'/60'/0'/0/0";
       const tx = '';
-      await expect(
-        bridge.deviceSignTransaction({
+      expect(
+        await bridge.deviceSignTransaction({
           hdPath,
           tx,
         }),
-      ).rejects.toThrow(Error);
-      expect(transportMiddlewareGetEthAppSpy).toHaveBeenCalledTimes(0);
-      expect(mockEthApp.signTransaction).toHaveBeenCalledTimes(0);
+      ).toBeUndefined();
+      expect(transportMiddlewareGetEthAppSpy).toHaveBeenCalledTimes(1);
+      expect(mockEthApp.clearSignTransaction).toHaveBeenCalledTimes(1);
```

Comment:
Should we still check for the inputs here? To my understanding, the `undefined` being returned is coming from `clearSignTransaction` itself, so we should also check for the inputs IMO?

```suggestion
      expect(mockEthApp.clearSignTransaction).toHaveBeenCalledTimes(1);
      expect(mockEthApp.clearSignTransaction).toHaveBeenCalledWith(hdPath, tx, {
        externalPlugins: true,
        erc20: true,
        nft: true,
      });
```

Reply:
yea, i agree with you, probablly add a check against the parameter is better tests.